public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/SimpleCondition.tsx

```diff
@@ -71,24 +73,37 @@ export const SimpleConditionEditor = ({
     updateThresholdFunction(value.value ?? EvalFunction.IsAbove, expressionQueriesList, dispatch);
   };
 
-  const onEvaluateValueChange = (event: FormEvent<HTMLInputElement>, index?: number) => {
-    if (isRange) {
+  const onEvaluateValueChange = (event: FormEvent<HTMLInputElement>, index = 0) => {
+    const value = event.currentTarget.value;
+
+    // Allow empty value
+    if (value === '') {
       const newParams = produce(simpleCondition.evaluator.params, (draft) => {
-        draft[index ?? 0] = parseFloat(event.currentTarget.value);
+        draft[index ?? 0] = undefined;
       });
-      // update the condition kept in the parent
       onChange({ ...simpleCondition, evaluator: { ...simpleCondition.evaluator, params: newParams } });
-      // update the reducer state where we store the queries
-      updateThresholdValue(parseFloat(event.currentTarget.value), index ?? 0, expressionQueriesList, dispatch);
-    } else {
-      // update the condition kept in the parent
-      onChange({
-        ...simpleCondition,
-        evaluator: { ...simpleCondition.evaluator, params: [parseFloat(event.currentTarget.value)] },
+      return;
+    }
+    // Allow '-' for negative numbers
+    if (value === '-') {
+      const newParams = produce(simpleCondition.evaluator.params, (draft) => {
+        draft[index ?? 0] = '-';
       });
-      // update the reducer state where we store the queries
-      updateThresholdValue(parseFloat(event.currentTarget.value), 0, expressionQueriesList, dispatch);
+      onChange({ ...simpleCondition, evaluator: { ...simpleCondition.evaluator, params: newParams } });
+      return;
     }
+
+    // Handle numeric values
+    const numericValue = parseFloat(value);
+    if (isNaN(numericValue)) {
+      return;
+    }
+
+    const newParams = produce(simpleCondition.evaluator.params, (draft) => {
```

Comment:
We already use immer here, so maybe it would be easier to do 
```suggestion
    const newParams = produce(simpleCondition, (draft) => {
      draft.evaluator.params[index] = numericValue;
      });
```
Then, we wouldn't need to create one more new object for the `onChange` below